docs/explanation/signature_backend.md

```diff
@@ -0,0 +1,129 @@
+# Signature backends
+
+To manage signatures, Joanie relies on signature providers. That means that
+we have to implement interfaces to interact with these providers. This is the
+purpose of the `BaseSignatureBackend` class which offers methods that are available
+when a signature procedure of a file has to be submitted, get the invitation link to
+sign the file, delete a signature procedure and handle incoming notifications
+from the signature provider. In this way, when implementing a new interface,
+we only need to focus on the logic to normalize data from the signature provider
+to Joanie.
+
+We start with the assumption that modern e-signature providers are based on
+notification webhook system. So Joanie relies on this feature to be notified
+when something happens on the signature provider side for a given file (signing finished, refusing
+to sign, ...)
+
+## Reference
+
+All signature backends are declared within `joanie.signature.backends` and must inherit from
+`BaseSignatureBackend`. This class implements 4 generic methods in charge of confirming a contract
+when it is signed, resetting a contract when it is refused, and there are two useful methods
+in getting the setting name and to retrieve its value to configure gracefully each signature
+backend classes.
+
+- **`get_setting_name(cls, name)`**
+
+- **`get_setting(cls, name)`**
+
+- **`confirm_signature(self, reference_id)`**
+
+- **`reset_contract(self, reference_id)`**
+
+On the other hand, your signature backend has to implement 4 methods :
+
+- **`submit_for_signature(self, definition: str, file_bytes: bytes, owner)`**
+
+- **`get_signature_invitation_link(self, recipient_email: str, reference_ids: list)`**
+
+- **`delete_signing_procedure(self, reference: str)`**
+
+- **`handle_notification(self, request)`**
+
+## Supported providers
+
+You can find all signature backends at `src/backend/joanie/signature/backends`.
+
+Currently, Joanie supports :
+
+- [LexPersona](https://www.lex-persona.com/)
+
+## How to
+
+### Use signature module from local environment
+
+To work on Joanie in a local environment, we suggest two solutions :
+
+#### Use the `DummySignatureBackend`
+
+In case you do not need to interact with a signature provider, we implemented
+a `DummySignatureBackend` which allows you to use Joanie locally without any
+configuration.
+
+#### Use the `LexPersonaClient`
+
+In case you want to use this class to interact with the signature provider, make
+sure that you have added all required configuration settings in your environment
+variables. Here is a list of **all the required configuration** you will need to set and their explanation, **once you have subscribed with Lex Persona** :
+
+* `JOANIE_SIGNATURE_BACKEND` : Declare the backend signature to use.
```

Comment:
In signature backend, there are also common settings which are optional but I think that could be explicit in this documentation.

- `JOANIE_SIGNATURE_VALIDITY_PERIOD`
- `JOANIE_SIGNATURE_INVITE_PERIOD`
- `JOANIE_SIGNATURE_TIMEOUT`